packages/core/execution-context/core-execution-context-common/src/types.ts

```diff
@@ -27,6 +27,8 @@ export type KibanaExecutionContext = {
   readonly description?: string;
   /** in browser - url to navigate to a current page, on server - endpoint path, for task: task SO url */
   readonly url?: string;
+  /** Metadata attached to the field. An optional parameter that allows to describe the execution context in more detail. **/
+  readonly meta?: { [key: string]: string | number | boolean | undefined };
```

Comment:
Given this is the only `core` file modified in this PR, I'm assuming this addition is not used by our EC system. May I ask what the purpose of this new field is, exactly? How/Where are you using it?

Reply:
we have 2 telemetries for which we need to pass additional meta information.  It's a `viewMode` and `fullscreenMode`. It this PR we set it in in `dashboard/public/application/hooks/use_dashboard_app_state.ts`

Reply:
Ok, I'm probably missing something here.

To my knowledge, apart from sending it to the server / ES for context tracking, the EC is used in 2 places:

1. By APM to add some labels:

https://github.com/elastic/kibana/blob/f609f092a5c179e2551cc68b97b72e97559a7b25/src/core/public/apm_system.ts#L73-L78

https://github.com/elastic/kibana/blob/f609f092a5c179e2551cc68b97b72e97559a7b25/packages/core/execution-context/core-execution-context-browser-internal/src/execution_context_service.ts#L62-L68

Which, without changes, is not using this new `meta` field

2.  For EBT to enrich the context

https://github.com/elastic/kibana/blob/f609f092a5c179e2551cc68b97b72e97559a7b25/packages/core/execution-context/core-execution-context-browser-internal/src/execution_context_service.ts#L119-L129

Which, without changes, is not using this new `meta` field either.

So, where exactly is this new field consumed? Is that in custom telemetry collectors?

Reply:
@pgayvallet For now we don't send it out in custom telemetry collectors but pick it from the context in the expression renderer and pass it to ui counters. Definitely possible we want to add such an integration at a later point though.

Reply:
It was implemented here cause we need somehow to pass `viewMode` and `fullscreenMode` properties from the container (e.g. `DashboardContainer`) into `Render expression` function. Therefore, the main consumer of this is: [lens/public/embeddable/embeddable.tsx](https://github.com/elastic/kibana/pull/135615/files#diff-06b33562b1971e1caf72a3a592b2ecd93a4b7199fe9d1bf66f36bc9d906325c1) (see `getExecutionContextEvents` method).